src/Illuminate/Routing/ImplicitRouteBinding.php

```diff
@@ -20,35 +27,7 @@ class ImplicitRouteBinding
      */
     public static function resolveForRoute($container, $route)
     {
-        $parameters = $route->parameters();
-
-        foreach ($route->signatureParameters(UrlRoutable::class) as $parameter) {
-            if (! $parameterName = static::getParameterName($parameter->getName(), $parameters)) {
-                continue;
-            }
-
-            $parameterValue = $parameters[$parameterName];
-
-            if ($parameterValue instanceof UrlRoutable) {
-                continue;
-            }
-
-            $instance = $container->make(Reflector::getParameterClassName($parameter));
-
-            $parent = $route->parentOfParameter($parameterName);
-
-            if ($parent instanceof UrlRoutable && in_array($parameterName, array_keys($route->bindingFields()))) {
-                if (! $model = $parent->resolveChildRouteBinding(
-                    $parameterName, $parameterValue, $route->bindingFieldFor($parameterName)
-                )) {
-                    throw (new ModelNotFoundException)->setModel(get_class($instance), [$parameterValue]);
-                }
-            } elseif (! $model = $instance->resolveRouteBinding($parameterValue, $route->bindingFieldFor($parameterName))) {
-                throw (new ModelNotFoundException)->setModel(get_class($instance), [$parameterValue]);
-            }
-
-            $route->setParameter($parameterName, $model);
-        }
+        (new static($container))->resolveSignatureParameters($route);
```

Comment:
Why are you newing up the container here?

Reply:
This is to keep the API backwards-compatible. Right now, the method is called via:

```php
ImplicitRouteBinding::resolveForRoute($container, $route);
```

So this just updates things so that the static `resolveForRoute` call instantiates the class and calls `resolveSignatureParameters` immediately.

Reply:
But to be clear, it's not newing up a the container. It's newing up the `ImplicitRouteBinding` class and passing the container into it.